OpenRA.Mods.Common/Activities/Enter.cs

```diff
@@ -72,6 +77,11 @@ public override bool Tick(Actor self)
 			if (!IsCanceling && useLastVisibleTarget && lastState == EnterState.Entering)
 				Cancel(self, true);
 
+			if (wasMobileMoved && mobile != null && mobile.MoveResult == MoveResult.MovementStuck)
+				return true;
+			else
+				wasMobileMoved = false;
+
```

Comment:
@PunkPun makes a good point about the Enter activity, we don't want to give up if the transport isn't at shore yet. Or maybe the transport is at shore, but the player needs to delete a wall that's in the way.

But we still need to reduce the pathfinding spam. I think adding a delay between retry attempts make sense for this activity. If we tried a move, don't try another move unless it's been 30 ticks since the last attempt. Maybe with some slight randomness so lots of units don't all retry on the exact same tick.

For `Enter`, we are happy to retry forever.